src/bessel.jl

```diff
@@ -590,6 +590,30 @@ function bessely(n::Integer, x::BigFloat)
     return z
 end
 
+"""
+    sphericalbesselj(nu, x)
+
+Spherical bessel function of the first kind at order `nu`, ``j_ν(x)``. This is the non-singular
+solution to the radial part of the Helmholz equation in spherical coordinates.
+"""
+function sphericalbesselj(nu, x::T) where {T}
+    besselj_nuhalf_x = besselj(nu + 1//2, x)
```

Comment:
Wouldn't `nu + one(nu)/2` be better?

Reply:
Shouldn't the order not introduce any round off errors. Seems better to do `1//2` , I'd think.

Reply:
`one(T)/2` is exact in floating point arithmetic and is 10x faster:
```julia
julia> plushalf_rat(x) = x + 1//2
plushalf_rat (generic function with 1 method)

julia> plushalf(x) = x + one(x)/2
plushalf (generic function with 1 method)

julia> @btime plushalf_rat(5)
  33.693 ns (0 allocations: 0 bytes)
11//2

julia> @btime plushalf(5)
  0.035 ns (0 allocations: 0 bytes)
5.5
```

Reply:
Hm, I'm trying to remember why I did that, I think I was concerned about someone putting in something like `sphericalbesselj(3//2, x)` and then getting a rounding error, but I don't think that's a very realistic concern.

Reply:
Fixed